lighthouse-core/gather/gatherers/trace-elements.js

```diff
@@ -67,26 +130,32 @@ class TraceElements extends Gatherer {
     if (!loadData.trace) {
       throw new Error('Trace is missing!');
     }
-    const traceOfTab = TraceProcessor.computeTraceOfTab(loadData.trace);
-    const lcpEvent = traceOfTab.largestContentfulPaintEvt;
+    const {largestContentfulPaintEvt, mainThreadEvents} =
+      TraceProcessor.computeTraceOfTab(loadData.trace);
     /** @type {Array<number>} */
     const backendNodeIds = [];
 
-    const lcpNodeId = TraceElements.getNodeIDFromTraceEvent(lcpEvent);
+    const lcpNodeId = TraceElements.getNodeIDFromTraceEvent(largestContentfulPaintEvt);
+    const clsNodeIds = TraceElements.getCLSNodeIdsFromMainThreadEvents(mainThreadEvents);
     if (lcpNodeId) {
       backendNodeIds.push(lcpNodeId);
     }
+    backendNodeIds.push(...clsNodeIds);
     // DOM.getDocument is necessary for pushNodesByBackendIdsToFrontend to properly retrieve nodeIds.
     await driver.sendCommand('DOM.getDocument', {depth: -1, pierce: true});
     const translatedIds = await driver.sendCommand('DOM.pushNodesByBackendIdsToFrontend',
       {backendNodeIds: backendNodeIds});
 
     // Mark the LCP element so we can find it in the page.
-    await driver.sendCommand('DOM.setAttributeValue', {
-      nodeId: translatedIds.nodeIds[0],
-      name: LH_ATTRIBUTE_MARKER,
-      value: 'largest-contentful-paint',
-    });
+    for (let i = 0; i < backendNodeIds.length; i++) {
+      const metricName =
+        lcpNodeId === backendNodeIds[i] ? 'largest-contentful-paint' : 'cumulative-layout-shift';
```

Comment:
can this instead use the `lcpNodeId` and `clsNodeIds` directly, and explicitly check both cases such that if neither are true (for some reason–shouldn't happen logically) the backend node is ignored?

Reply:
bump on this